### DISCUSSION TITLE
Generate SVG not consistent ?

### DISCUSSION THREAD

**Discussion by haunt98** [caksoylar/keymap-drawer#41] [Q&A] [answered]
Hi @caksoylar, thanks for your wonderful tools. I have used it to draw my keymap for some time.

But after I editing config heavily. Each time I try to draw svg from yaml.
The generated svg is changed even if yaml stay the same.

You can try with my repo: https://github.com/haunt98/qmk_keymaps

The config is stores here:
- https://github.com/haunt98/qmk_keymaps/blob/main/dztech_dz60rgb_wkl/caksoylar_keymap_drawer/config.yaml
- https://github.com/haunt98/qmk_keymaps/blob/main/dztech_dz60rgb_wkl/caksoylar_keymap_drawer/keymap.yaml

The changed is showed in this commit: https://github.com/haunt98/qmk_keymaps/commit/f186a04e19213cf6321ab36e138cab88b999279c

As I understand, the order when generate svg is not consistent.

**Comment by caksoylar** [comment]
Just to confirm, the generated SVGs look identical right?

I think I know why this happens. A set data structure is used to look up all the glyphs used in the keymap, then we fetch the the SVG defs and write them. Set is not guaranteed to preserve item ordering so I guess the written order might change.

This shouldn't affect the visual but it is annoying to create unnecessary diffs. I will sort the glpyh SVGs before writing so the order doesn't change across runs.

**Reply by haunt98** [reply]
Yeah, it's as you say.
The generated SVGs look the same, only the order is changed each time I re-run.

**Comment by caksoylar** [comment, marked as answer]
This was because the SVG glyph definitions order in the output SVG file was not deterministic, it is resolved by https://github.com/caksoylar/keymap-drawer/commit/6e7ac694d427fbcf71846b31f1339ef044be9c6b.

**Reply by haunt98** [reply]
Thank you for this. I will try later :D

**Reply by haunt98** [reply]
Upgrading to v0.11.0 solve it. Thank @caksoylar! No more inconsistent everytime I re-draw.
